node/network/collator-protocol/src/collator_side.rs

```diff
@@ -695,7 +695,7 @@ pub(crate) async fn run(
 				let (relay_parent, validator_id, peer_id) = match res {
 					Some(res) => res,
 					// Will never happen, but better to be safe.
-					None => continue,
+					None => return Ok(()),
```

Comment:
this branch is unreachable as you can see from the comment above as well as by looking at the `impl stream::Stream for ConnectionRequests`. Also, the `.fuse()` above can be removed (I think it was at some point?).

Reply:
Still, if the behavior of `ConnectionRequests` did change at some point, I would rather have this line of code here than the old version.